programs/psa/key_ladder_demo.sh

```diff
@@ -17,9 +17,15 @@
 
 set -e -u
 
-program="${0%/*}"/key_ladder_demo
+program=./programs/psa/key_ladder_demo
```

Comment:
Unfortunately this breaks
```
cd programs/psa
./key_ladder_demo.sh
```
which is a perfectly reasonable to try to run the script. So this way is nice for our CI, but not nice for our users.

Reply:
That's true, I tried to mitigate that with an error message (could be reworded to be more clear). It's not ideal, but is it too much of a problem to require the scripts to be run from a certain place, provided that it's made clear to the user?

Alternatively, I could:
- Add some kind of `--build-dir` option to pass an out-of-tree path.
- Do some kind of detection magic and `cd` to the top of the tree at the start.
- Just ignore out-of-tree builds for the purposes of the demo scripts.

Reply:
Digging back, it looks like I implemented some kind of detection magic in #2698. How about building on that?